algolia-search/src/search-form.tsx

```diff
@@ -0,0 +1,158 @@
+
+    if (query) searchParams.set("key", query);
+    const pageTypes = pageTypeRefinements.filter(item => item.isRefined).map(item => item.value);
+    const sharedTypes = sharedRefinements.filter(item => item.isRefined).map(item => item.value);
+
+    if (pageTypes.length >= 1) searchParams.set("page-type", pageTypes.join(','))
+    if (sharedTypes.length >= 1) searchParams.set("shared-type", sharedTypes.join(','))
+
+    window.history.replaceState(null, '', `?${searchParams.toString()}`);
+  }, [query, pageTypeRefinements, sharedRefinements])
+
+  return (
+    <form
+      role="search"
+      onSubmit={(e) => {
+        e.preventDefault();
+        refine(inputRef.current.value)
+      }}
+      onReset={(e) => {
+        inputRef.current.value = ""
+        refine("");
+        pageTypeRefinements.map(refinementItem => {
+          if (refinementItem.isRefined) refinePageTypes(refinementItem.value)
+        })
+        sharedRefinements.map(refinementItem => {
+          if (refinementItem.isRefined) refineSharedTypes(refinementItem.value)
+        })
+      }}
+    >
+      <div>
+        <label htmlFor="keyword-search-algolia">
+          Keywords<span className="visually-hidden">&nbsp;Search</span>
+        </label>
+        <input
+          id="keyword-search-algolia"
+          ref={inputRef}
+          autoComplete="on"
+          autoCorrect="on"
+          autoCapitalize="off"
+          spellCheck={true}
+          maxLength={128}
+          type="search"
+          defaultValue={query}
+        />
+
+        <div style={{display: "flex", gap: "1rem", marginTop: "1rem"}}>
+          <button type="submit">Submit</button>
+
+        </div>
+      </div>
+
+      <div style={{float: "left", width: "33%", "margin-top": "39px"}}>
+        <fieldset>
+          <legend>Basic Page Types</legend>
+
+          <ul style={{listStyle: "none", paddingLeft: "0", marginInline: "0"}}>
+            {pageTypeRefinements.map((item, i) =>
+              <li
+                key={i}
+                className="w-fit flex items-center justify-center border border-black p-5 mb-5 last:mb-0"
+                style={{
+                  alignItems: "center",
+                }}
+              >
+                <label style={{
+                  marginTop: "1rem",
+                  display: "flex",
+                  alignItems: "center",
+                }}>
+                  <input
+                    type="checkbox"
+                    onChange={() => refinePageTypes(item.value)}
+                    checked={item.isRefined}
+
+                  />
+                  {item.value} ({item.count})d
+                </label>
+              </li>
+            )}
+          </ul>
+        </fieldset>
+        <fieldset>
+          <legend>Shared Types</legend>
+
+          <ul style={{listStyle: "none", paddingLeft: "0", marginInline: "0"}}>
+            {sharedRefinements.map((item, i) =>
+              <li
+                key={i}
```

Comment:
You need to have unique keys. You can't use the same key as a previous map. something like 
```
`shared-${i}`
```